src/neo-vm/Types/Integer.cs

```diff
@@ -55,5 +53,10 @@ public override int GetByteLength()
                 _length = value.GetByteCount();
             return _length;
         }
+
+        internal override ReadOnlyMemory<byte> ToMemory()
+        {
+            return value.IsZero ? ReadOnlyMemory<byte>.Empty : value.ToByteArray();
```

Comment:
Maybe we should cache the `ToByteArray` the first time that will be used, what do you think?